packages/desktop-client/src/components/schedules/LinkSchedule.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useRef, useState } from 'react';
+
+import { useSchedules } from 'loot-core/src/client/data-hooks/schedules';
+import { send } from 'loot-core/src/platform/client/fetch';
+
+import { type BoundActions } from '../../hooks/useActions';
+import Modal from '../common/Modal';
+import Search from '../common/Search';
+import Text from '../common/Text';
+import View from '../common/View';
+import { type CommonModalProps } from '../Modals';
+
+import { ROW_HEIGHT, SchedulesTable } from './SchedulesTable';
+
+export default function ScheduleLink({
+  modalProps,
+  actions,
+  transactionIds: ids,
+}: {
+  actions: BoundActions;
+  modalProps?: CommonModalProps;
+  transactionIds: string[];
+}) {
+  let [filter, setFilter] = useState('');
+
+  let scheduleData = useSchedules({
+    transform: useCallback((q: any) => q.filter({ completed: false }), []),
```

Comment:
Used `any` as a last resort here. 

@albertogasparin Can you offer any insight as to a more appropriate type for `q` here, given the expected type for the `transform` function [from schedules.tsx](https://github.com/actualbudget/actual/blob/bug/1500-empty-schedules-modal/packages/loot-core/src/client/data-hooks/schedules.tsx#L21)?

Reply:
I'm pretty sure that the actual fix is

```diff
diff --git a/packages/loot-core/src/client/data-hooks/schedules.tsx b/packages/loot-core/src/client/data-hooks/schedules.tsx
index 61e24192..3603561f 100644
--- a/packages/loot-core/src/client/data-hooks/schedules.tsx
+++ b/packages/loot-core/src/client/data-hooks/schedules.tsx
@@ -1,5 +1,6 @@
 import React, { createContext, useEffect, useState, useContext } from 'react';
 
+import { type Query } from '../../shared/query';
 import { getStatus, getHasTransactionsQuery } from '../../shared/schedules';
 import q, { liveQuery } from '../query-helpers';
 
@@ -18,7 +19,7 @@ function loadStatuses(schedules, onData) {
   });
 }
 
-type UseSchedulesArgs = { transform?: <T>(v: T) => T };
+type UseSchedulesArgs = { transform?: (q: Query) => Query };
 export function useSchedules({ transform }: UseSchedulesArgs = {}) {
   let [data, setData] = useState(null);
 
```

Then you can have the `useCallback` call either un-annotated or be `(q: Query)`.

Reply:
I like it, @kyrias. Thanks.
Looked around and `Query` works for everywhere we use `useSchedules`.
Annnnd, the usage in EditRule.js would make TS mad just like this one did when I first converted it.